src/libs/actions/IOU.ts

```diff
@@ -675,11 +675,11 @@ function buildOnyxDataForMoneyRequest(
     ];
 
     // Policy won't be set for P2P cases for which we don't need to compute violations
-    if (!policy?.id) {
+    if (policy && PolicyUtils.isPaidGroupPolicy(policy)) {
```

Comment:
Is this condition backwards?
```suggestion
    if (!policy || !PolicyUtils.isPaidGroupPolicy(policy)) {
```